### PR TITLE
Add SupplementalFile ingest API

### PR DESCRIPTION
**Pull request by masaball**
Related issue: #5801

### CODE REVIEW

**Review by cjcolvar — COMMENTED** [review]
I have some ideas to make some of this simpler and align more with the other API endpoints' JSON format.  I know you've tried a lot of things to get to this point and you may have already tried what I'm thinking.  Could we talk this out synchronously since I think that'll be easier than in comments here.

Good work on this so far!  This is tricky with file upload being new to the API!

File: app/controllers/supplemental_files_controller.rb [hunk 1]
```diff
@@ -30,65 +33,105 @@ class SupplementalFilesController < ApplicationController
...
     raise Avalon::SaveError, @object.errors[:supplemental_files_json].full_messages unless @object.save
 
     flash[:success] = "Supplemental file successfully added."
 
     respond_to do |format|
-      format.html { redirect_to edit_structure_path }
-      format.json { head :created, location: object_supplemental_file_path }
+      format.html { 
+        if request.headers['Avalon-Api-Key'].present?
+          render json: { supplemental_file: @supplemental_file.id }, status: :created
```
[the remark this answers is not in the document]
Comment (cjcolvar) [reply app/controllers/supplemental_files_controller.rb:78]: Thanks for explaining this case!  I understand now about how this is the binary file only upload path.  Could you please leave a line or two of inline comments to explain it here?

File: app/models/supplemental_file.rb [hunk 2]
```diff
@@ -64,6 +63,25 @@ def caption_transcript?
     tags.include?('caption') && tags.include?('transcript')
   end
 
+  def as_json(options={})
+    type = if tags.include?('caption')
+             'caption'
+           elsif tags.include?('transcript')
+             'transcript'
+           else
+             'generic'
```
[the remark this answers is not in the document]
Comment (cjcolvar) [reply app/models/supplemental_file.rb:72]: After chatting, I understand how you're mapping type to tags and back again based upon the api spec in the ticket.  Using `type` in the JSON probably does make the response clearer than the grab bag of terms in `tags`.  In the future we could revisit if we have a clear enough understanding of the shape of supplemental files that we could add a `type` field in the model (and `machine_generated`, etc.) and move away from `tags` as the freeform way for assigning behavior.

**Review by cjcolvar — COMMENTED** [review]
Thanks for taking the time to chat about this!

File: app/controllers/supplemental_files_controller.rb [hunk 3]
```diff
@@ -14,6 +14,9 @@
 
 # frozen_string_literal: true
 class SupplementalFilesController < ApplicationController
+  include Rails::Pagination
+
+  before_action :authenticate_user!, except: [:show, :captions]
```
[the remark this answers is not in the document]
Comment (masaball) [reply app/controllers/supplemental_files_controller.rb:19]: Looks like you are right that this shouldn't be necessary. Good catch.

**Review by cjcolvar — APPROVED** [review]
The changes look great! :tada:

**Pull request merged**